internal/handlers/msg_comparison_test.go

```diff
@@ -0,0 +1,430 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"math"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
+)
+
+const (
+	DoubleDataType                 = "double"
+	DoubleZeroDataType             = "double-zero"
+	DoubleNegativeInfinityDataType = "double-negative-infinity"
+	DoublePositiveInfinityDataType = "double-positive-infinity"
+	ArrayDataType                  = "array"
+	EmptyArrayDataType             = "array-empty"
+	DoubleNanDataType              = "double-nan"
+	DoubleMaxDataType              = "double-max"
+	DoubleSmallestDataType         = "double-smallest"
+	StringDataType                 = "string"
+	EmptyStringDataType            = "string-empty"
+	Int32DataType                  = "int32"
+	Int32ZeroDataType              = "int32-zero"
+	Int32MaxDataType               = "int32-max"
+	Int32MinDataType               = "int32-min"
+	Int64DataType                  = "int64"
+	Int64ZeroDataType              = "int64-zero"
+	Int64MaxDataType               = "int64-max"
+	Int64MinDataType               = "int64-min"
+	TimestampDataType              = "timestamp"
+	SchemaWithAllTypes             = "values"
+	CollectionWithAllTypes         = "values"
+	BinaryDataType                 = "binary"
+	EmptyBinaryDataType            = "binary-empty"
+	BoolFalseDataType              = "bool-false"
+	BoolTrueDataType               = "bool-true"
+	DateTimeDataType               = "datetime"
+	DateTimeEpochDataType          = "datetime-epoch"
+	DateTimeMinYearDataType        = "datetime-year-min"
+	DateTimeMaxYearDataType        = "datetime-year-max"
+)
+
+func TestComparison(t *testing.T) {
```

Comment:
Let's integrate it into `TestFind`

Reply:
I'll do it.
Should I leave a `findValueByComparisonOperator` func?